### ISSUE TITLE
react/display-name doesn't trigger when functional compo includes named function

### ISSUE THREAD

**Issue by dwelle** [yannickcr/eslint-plugin-react#2221]
When a functional component includes a named function in its body, the `react/display-name` rule does not trigger for the wrapping component.

```js
export default function () {

    function fetchData () {} // <-- causes react/display-name not to trigger

    return <div/>;
}
```

- eslint: `5.15.3`
- eslint-plugin-react: `7.12.4`

**Comment by ljharb** [comment]
Although I'd suggest never creating a function inside an SFC (inside the render path), this definitely sounds like a bug.

**Comment by dwelle** [comment]
When using hooks, you almost exclusively create functions inside FCs.

**Comment by ljharb** [comment]
That doesn't mean it's a good idea, it just means hooks might encourage that pattern.
